services/xero/create_bulk_invoices.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module Xero
+  class CreateBulkInvoices
+    extend ::Dry::Initializer
+
+    BATCH_SIZE = 50  # Max 50 invoices per request
+    RATE_LIMIT = 60  # 60 API calls per minute
+
+    option :invoices, model: Invoice
+
+    def call
+      @invoices.each_slice(BATCH_SIZE).with_index do |invoice_batch, index|
+        # Handle rate limit by pausing after every RATE_LIMIT number of API calls
+        sleep(60) if index > 0 && (index % RATE_LIMIT == 0)
+        response = Xero::Api::Accounting::CreateInvoices.new(invoices: invoice_batch).call
+
+        # Update the local invoices with the xero_id from the response
+        update_local_invoices(invoice_batch, response)
+      end
+    end
```

Comment:
```suggestion
    def call
      invoices.find_in_batches(batch_size: 50) do |batch|
        response = Xero::Api::Accounting::CreateInvoices.new(invoices: batch).call
        attributes = batch.map do |invoice, index|
          xero_invoice =  response.invoices.select { |res| res[:reference_number] == invoice.id }
          { id: invoice.id, xero_id: xero_invoice.invoice_id }
        end
        
        Invoice.upsert_all(attributes, update_only: :xero_id, unique_by: :id)
      end
    end
```

Pas create coba asupkeun reference number / invoice id meh teu pake index, jadi lewih consistent @persada21